### ISSUE TITLE
Send a notification when a long-running command finishes

### ISSUE THREAD

**Issue by txtyash** [nushell/nushell#6364]
### Related problem

There's a plugin for fish shell called "done" which notifies the user when a long job is complete. A similar feature would be a nice to have.

### Describe the solution you'd like

Built-in notification feature that notifies the user whenever certain jobs like `apt install` are complete.

**Comment by fdncred** [comment]
yes, it would be nice to have. nushell doesn't currently support jobs.

**Comment by 132ikl** [comment]
You should be able to do this with [hooks](https://www.nushell.sh/book/hooks.html):

```nushell
const NOTIFY_THRESHOLD = 1sec

$env.config.hooks.pre_execution ++= [{ $env.NOTIFY_START = date now }]
$env.config.hooks.pre_prompt ++= [{
  let start = $env.NOTIFY_START? | default (date now)
  if ((date now) - $start) > $NOTIFY_THRESHOLD {
    notify-send "Job done"
  }
}]
```

You could also try extracting the command from `history` and display that in the notification as well.

If this meets your needs, let me know and I'll close this issue.

**Comment by 132ikl** [comment]
hi, I'm going to close this issue. if the above solution doesn't solve the problem, let me know and I'll re-open the issue.
